e2e-tests/gef/cypress/e2e/pages/bank-review-date.js

```diff
@@ -9,6 +11,16 @@ const bankReviewDate = {
   bankReviewDateDetails: () => cy.get('[data-cy="bank-review-date-details"]'),
   continueButton: () => cy.get('[data-cy="continue-button"]'),
   saveAndReturnButton: () => cy.get('[data-cy="save-and-return-button"]'),
+
+  /**
+   * Fill in the bank review date form
+   * @param {Date} date
+   */
+  fillInBankReviewDate: (date) => {
+    this.bankReviewDateDay().clear().type(format(date, 'd'));
+    this.bankReviewDateMonth().clear().type(format(date, 'M'));
+    this.bankReviewDateYear().clear().type(format(date, 'yyyy'));
+  },
```

Comment:
Is this pattern (putting functions into `page` selectors), something we have in other places?

I'm not a fan of this because:

1) Page selectors are no longer pure selectors
2) Individual tests have to import the page in order to call the function
3) Makes it difficult to move to another place in the future

Unless this is a pattern in other places, could we move this to a non-page specific location, i.e a standalone cypress command? So it can be called like this for example:

```js
cy.completeBankReviewDate();
```

"fillIn" or "complete" or something else is up for debate I think.

Reply:
I'd seen reusable commands requested on one of Beths PRs & this was how she resolved it (see navigateThroughFacilityEndDateAmendmentPages in e2e-tests\tfm\cypress\e2e\pages\amendments\amendmentsPage.js). I agree that these pages files should be pure & only contain selectors, but I'm not much more of a fan of commands on the root level cy object. Will implement for now, but think a nicer pattern we could adopt would be importing these from a page-specific helpers file

Reply:
It's an interesting topic @MarRobSoftwire - the problem with having such functions in a page file is that, it's no longer "a page", if that makes sense. Also, it means anytime you want to run such a function, you have to import it. There could be 5-10+ more instances in an E2E suite where we want to e.g complete and submit X form, having this as a root command would be cleaner and DRYer

It's open for discussion, there's a few ways to do such things but I think top level cy commands are the cleanest way to go, as long as they're named and structured well